lib/src/create_report.dart

```diff
@@ -755,6 +756,83 @@ Future<ReportSection> _trustworthyDependency(PackageContext context) async {
           _makeSummary([dependencySection, sdkSection], basePath: packageDir));
 }
 
+_Issue _explanationToIssue(Explanation explanation) =>
+    _Issue(explanation.finding, suggestion: explanation.explanation);
+
+Future<ReportSection> _nullSafety(String packageDir, Pubspec pubspec) async {
+  // TODO(sigurdm): Currently we don't give any points for null-safety.
+  // Raise this after null-safety beta.
+  const maxPoints = 0;
+
+  _Subsection subsection;
+  if (File(p.join(packageDir, '.dart_tool', 'package_config.json'))
+      .existsSync()) {
+    final tagger = Tagger(packageDir);
+
+    final nullSafetyTags = <String>[];
+    final explanations = <Explanation>[];
+    tagger.nullSafetyTags(nullSafetyTags, explanations);
+    if (pubspec.sdkConstraintStatus.hasOptedIntoNullSafety) {
+      if (nullSafetyTags.contains('is:null-safe')) {
+        subsection = _Subsection(
+            'Package and dependencies are fully migrated to null-safety',
+            explanations.map(_explanationToIssue).toList(),
+            maxPoints,
+            maxPoints,
+            _Status.good);
+      } else {
+        subsection = _Subsection(
+            'Package declares support for null-safety, but there are issues',
+            [
+              ...explanations.map(_explanationToIssue).toList(),
+              _Issue('For more information',
+                  suggestion: 'Try running `pub outdated --mode=null-safety`.\n'
+                      'Be sure to read the [migration guide](https://dart.dev/null-safety/migration-guide).')
+            ],
+            0,
+            maxPoints,
+            _Status.bad);
+      }
+    } else {
+      subsection = _Subsection(
+          'Package does not opt in to  null-safety',
+          [
+            _Issue(
+              'Package version constraint lower bound is below 2.12.0.',
```

Comment:
nit: can we use a const for `2.12.0`, maybe reuse `_firstNullSafetyVersion` for it?